library/md.c

```diff
@@ -244,33 +249,35 @@ int mbedtls_md_setup( mbedtls_md_context_t *ctx, const mbedtls_md_info_t *md_inf
 
 int mbedtls_md_starts( mbedtls_md_context_t *ctx )
 {
-    if( ctx == NULL || ctx->md_info == NULL )
-        return( MBEDTLS_ERR_MD_BAD_INPUT_DATA );
+    MBEDTLS_MD_VALIDATE_RET( ctx != NULL );
+    MBEDTLS_MD_VALIDATE_RET( ctx->md_info != NULL );
```

Comment:
Non-blocking: In line with the other PRs, I'd prefer to go with the first non-NULL check only.